src/swaps/hop/HopSwapProvider.ts

```diff
@@ -0,0 +1,579 @@
+    const client = this._getClient(network, walletId, from, fromAccountId)
+    const privKey = await client.wallet.exportPrivateKey()
+    return new Wallet(privKey, provider)
+  }
+
+  async _getBridgeWithSigner(hopAsset: any, hopChainFrom: any, network: Network, walletId: string, from: string, fromAccountId: string) {
+    const chainFrom = this.getChain(hopChainFrom.slug)
+    const client = this._getClient(network, walletId, from, fromAccountId)
+    const privKey = await client.wallet.exportPrivateKey()
+    const hop = this._getHop(network)
+    const signer = new Wallet(privKey, hop?.getChainProvider(chainFrom))
+    const bridge = hop?.connect(signer).bridge(hopAsset)
+    return bridge
+  }
+
+  _findAsset(asset: any, chain: any, tokens: any, tokenName: string) {
+    if (asset.type === 'native') {
+      // native asset
+      if (
+        // this.getToken[asset.code] === tokenName ||
+        // this.getToken[asset.matchingAsset] === tokenName
+        asset.code === tokenName ||
+        asset.matchingAsset === tokenName
+      ) {
+        return tokenName
+      }
+    } else {
+      // erc20 asset
+      if (
+        tokens[chain]?.l1CanonicalToken?.toLowerCase() === asset?.contractAddress.toLowerCase() ||
+        tokens[chain]?.l2CanonicalToken?.toLowerCase() === asset?.contractAddress.toLowerCase()
+      ) {
+        return tokenName
+      }
+    }
+  }
+
+  _getSendInfo(assetFrom: any, assetTo: any, hop: any) {
+    if (!assetFrom || !assetTo) return null
+    const _chainFrom = this.getChain(assetFrom.chain)
+    const _chainTo = this.getChain(assetTo.chain)
+    if (!_chainFrom || !_chainTo) return null
+    const availableToken = this._getAllTokens(hop)
+    let _from, _to
+    for (const token in availableToken) {
+      if (!_from) _from = this._findAsset(assetFrom, _chainFrom.slug, availableToken[token], token)
+      if (!_to) _to = this._findAsset(assetTo, _chainTo.slug, availableToken[token], token)
+    }
+    if (!_from || !_to || _from !== _to) return null
+    const supportedAssetsFrom = hop.getSupportedAssetsForChain(_chainFrom.slug)
+    const supportedAssetsTo = hop.getSupportedAssetsForChain(_chainTo.slug)
+    if (!supportedAssetsFrom[_from] || !supportedAssetsTo[_to]) return null
+    return { bridgeAsset: _from, chainFrom: _chainFrom, chainTo: _chainTo }
+  }
+
+  // eslint-disable-next-line no-unused-vars
+  public async getQuote({ network, from, to, amount } : QuoteRequest) {
+    if (amount <= new BN(0)) return null
+    const assetFrom = cryptoassets[from]
+    const assetTo = cryptoassets[to]
+    const fromAmountInUnit = currencyToUnit(cryptoassets[from], new BN(amount))
+    const hop = this._getHop(network)
+    if (!hop || !hop.isValidChain(assetFrom.chain) || !hop.isValidChain(assetTo.chain)) return null
+    const info = this._getSendInfo(assetFrom, assetTo, hop)
+    if (!info?.bridgeAsset || !info?.chainFrom || !info?.chainTo) return null
```

Comment:
Isn't all these already checked in getSendInfo and returning null? Why are we checking each property again? And they shouldn't be optional then if they are required for the swap.